zkllvm.nix

```diff
@@ -0,0 +1,50 @@
+{ lib,
+  stdenv,
+  src_repo,
+  pkgs,
+  ninja,
+  pkg-config,
+  cmake,
+  boost183,
+  # We'll use boost183 by default, but you can override it
+  boost_lib ? boost183,
+  gdb,
+  cmake_modules,
+  crypto3,
+  blueprint,
+  enableDebugging,
+  enableDebug ? false,
+  runTests ? false,
+  }:
+let
+  inherit (lib) optional;
+in stdenv.mkDerivation {
+  name = "Zkllvm";
+  src = src_repo;
+
+  nativeBuildInputs = [ cmake ninja pkg-config ] ++ (lib.optional (!stdenv.isDarwin) gdb);
+
+  # enableDebugging will keep debug symbols in boost
+  propagatedBuildInputs = [ (if enableDebug then (enableDebugging boost_lib) else boost_lib) ];
+
+  buildInputs = [crypto3 blueprint cmake_modules pkgs.python3 pkgs.git pkgs.cargo pkgs.openssl];
+
+  cmakeFlags =
+    [
+      (if runTests then "-DBUILD_TESTS=TRUE" else "")
+      (if runTests then "-DCMAKE_ENABLE_TESTS=TRUE" else "")
+      (if enableDebug then "-DCMAKE_BUILD_TYPE=Debug" else "-DCMAKE_BUILD_TYPE=Release")
+      (if enableDebug then "-DCMAKE_CXX_FLAGS=-ggdb" else "")
+      "-DZKLLVM_VERSION=0.1.18"
+      "-DGENERATE_EVM_VERIFIER=TRUE"
+    ];
+
+  ninjaFlags = "assigner clang transpiler";
+
+  doCheck = false;
+  doInstall = false;
```

Comment:
Why don't we install?

Reply:
It fails to install to the system directories

Reply:
Thus, final binaries can't be used as a derivation output?

Reply:
Yes, I have, here is what happening:
```   
    > [713/714] Install the project...
       > -- Install configuration: "Debug"
       > -- Installing: /nix/store/hplw610rmlj1gs1crk6z9fq31ha5q0ly-Zkllvm/share/zkllvm/CircuitCompile.cmake
       > CMake Error at cmake_install.cmake:58 (file):
       >   file cannot create directory: /etc/ld.so.conf.d.  Maybe need administrative
       >   privileges.
       >
       >
       > FAILED: CMakeFiles/install.util
       > cd /build/source/build && /nix/store/ll2b6cslly4v30im7qk34f9y21ziiz1i-cmake-3.29.6/bin/cmake -P cmake_install.cmake
       > ninja: build stopped: subcommand failed.
```

Reply:
Absolute path [here](https://github.com/NilFoundation/zkLLVM/blob/64b76de21aa318b3772f20fdab11cc823eab4ad2/CMakeLists.txt#L144) does not respect `CMAKE_INSTALL_PREFIX` flag. You could try to make it relative (remove first `/`)

Reply:
How about using `CMAKE_INSTALL_SYSCONFDIR` from `GNUInstallDirs`?

```cmake
install(FILES "${CMAKE_CURRENT_BINARY_DIR}/zkllvm.conf" DESTINATION "${CMAKE_INSTALL_SYSCONFDIR}/d.so.conf.d")
```

I guess we need to check, that it will work with CPack properly.

Reply:
Thanks for suggestion, I have applied it here
https://github.com/NilFoundation/zkLLVM/pull/614/commits/deba633744f206a50351169e3f42d687a5cf2068

now install and fixup phases work as expected
https://github.com/NilFoundation/zkLLVM/pull/614/commits/9fedec9082fcef038e5ad71259bb28980ca44f1a